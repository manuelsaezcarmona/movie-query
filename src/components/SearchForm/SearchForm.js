import styles from './styles.module.css';

export default function SearchForm() {
  return (
    <div className={styles.search}>
      <form action="" className={styles.search__form}>
        <input className={styles.search__inputtext} type="text" name="search" placeholder="Busca tu pelicula" />
        <button className={styles.search__button} type="button">
          Busca
        </button>
      </form>
    </div>
  );
}
