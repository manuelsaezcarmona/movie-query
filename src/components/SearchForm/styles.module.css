.search {
  margin-bottom: 16px;
}

.search__form {
  display: flex;
  justify-content: space-around;
  flex-direction: column;
  align-items: center;
}

.search__inputtext {
  border: 1px solid grey;
  border-radius: 8px;
  box-sizing: border-box;
  color: --navbar-color;
  font-size: 16px;
  font-weight: bold;
  height: 40px;
  outline: none;
  padding: 16px;
  width: 100%;
  -webkit-box-shadow: var(--shadow-card);
  -moz-box-shadow: var(--shadow-card);
  box-shadow: var(--shadow-card);
}

.search__button {
  color: var(--text-color);
  padding: 1em;
  margin: 0.7em;
  font-family: var(--navbar-font);
  border-radius: 5px;
  border: 1px #dcdcdc solid;
  background: linear-gradient(180deg, #ededed 5%, #dfdfdf 100%);
  text-shadow: 1px 1px 1px #ffffff;
  box-shadow: inset 1px 1px 2px 0px #ffffff;
  cursor: pointer;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 40px;
}
.search__button:hover {
  background: linear-gradient(180deg, #dfdfdf 5%, #ededed 100%);
}

.search__icon {
  margin: 10px 18px;
  font-size: 24px;
}

@media (min-width: 769px) {
  .search__form {
    flex-direction: row;
  }
}
