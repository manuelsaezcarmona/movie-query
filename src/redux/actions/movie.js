import { getMoviesByTerm } from '../../services/http';
import { types } from '../actiontypes';

export const getAllMovies = (result) => ({
  type: types.MOVIE_GET_ALLMOVIES,
  payload: result
});

export const startGetAllMovies = (term, page) => async (dispatch) => {
  try {
    const result = await getMoviesByTerm(term, page);

    if (!result.success) {
      return result.status_message;
    }
    dispatch(getAllMovies(result));
    return result;
  } catch (error) {
    return error;
  }
};
