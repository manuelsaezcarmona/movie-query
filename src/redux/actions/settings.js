import { types } from '../actiontypes';

export const previousPage = (page) => {
  const decrementedPage = page - 1;
  return { type: types.SETTINGS_SET_CURRENTPAGE, payload: decrementedPage };
};

export const nextPage = (page) => {
  const incrementedPage = page + 1;
  return { type: types.SETTINGS_SET_CURRENTPAGE, payload: incrementedPage };
};

export const setCurrentPage = (page) => ({
  type: types.SETTINGS_SET_CURRENTPAGE,
  payload: page
});

export const setCurrentSearch = (searchFilm) => ({
  type: types.SETTINGS_SET_CURRENT_SEARCH,
  payload: searchFilm
});
