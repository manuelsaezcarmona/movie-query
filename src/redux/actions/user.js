import { types } from '../actiontypes';

export const addFavourite = (movie) => ({
  type: types.USER_ADD_FAVOURITES,
  payload: movie
});

export const deleteFavourite = (movieID) => ({
  type: types.USER_REMOVE_FAVOURITES,
  payload: movieID
});
