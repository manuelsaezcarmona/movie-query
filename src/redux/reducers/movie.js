import { types } from '../actiontypes';

const initialState = {
  page: 0,
  movies: [],
  activemovie: {},
  totalpages: 0,
  detailedmovie: {}
};

export const movieReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.MOVIE_GET_ALLMOVIES:
      return {
        ...state,
        page: action.payload.page,
        movies: action.payload.results,
        totalpages: action.payload.total_pages
      };
    case types.MOVIE_SET_ACTIVE_MOVIE:
      return {
        ...state,
        activemovie: action.payload
      };
    case types.MOVIE_SET_DETAILED_MOVIE: {
      // console.log(action.payload);
      return {
        ...state,
        detailedmovie: action.payload
      };
    }
    default:
      return state;
  }
};
