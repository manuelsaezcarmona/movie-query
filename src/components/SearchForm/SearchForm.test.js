import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render, screen, fireEvent } from '@testing-library/react';
import SearchForm from './SearchForm';
import { startGetAllMovies } from '../../redux/actions/movie';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  user: {
    username: 'test',
    favourites: []
  },
  movie: {
    page: 1,
    movies: [],
    activemovie: {},
    totalpages: 3,
    detailedmovie: {}
  },
  settings: {
    currentpage: 1,
    searchfilm: 'Avengers'
  }
};

const store = mockStore(initialState);
store.dispatch = jest.fn();

jest.mock('../../redux/actions/movie', () => ({
  startGetAllMovies: jest.fn()
}));

describe('Given the SearchForm', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <SearchForm />
      </Provider>
    );
  });
  test('The component is rendered', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <SearchForm />
      </Provider>
    );
    expect(asFragment(<SearchForm />)).toMatchSnapshot();
  });
  describe('Given the Search Button', () => {
    test('The button must called to handleSubmit with text', () => {
      const search = 'something';
      const button = screen.getByRole('button', { options: { name: 'searchButton' } });
      const inputText = screen.getByRole('textbox');
      fireEvent.change(inputText, { target: { value: search } });

      fireEvent.click(button);
      expect(startGetAllMovies).toHaveBeenCalledWith(search, 1);
    });
  });
});
