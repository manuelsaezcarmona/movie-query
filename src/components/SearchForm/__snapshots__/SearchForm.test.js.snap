// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the SearchForm The component is rendered 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <form
      class="search__form"
    >
      <input
        class="search__inputtext"
        name="search"
        placeholder="Busca tu pelicula"
        type="text"
        value=""
      />
      <button
        class="search__button"
        name="searchButton"
        type="button"
      >
        <svg
          class="search__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.7 13.3l-3.81-3.83A5.93 5.93 0 0 0 13 6c0-3.31-2.69-6-6-6S1 2.69 1 6s2.69 6 6 6c1.3 0 2.48-.41 3.47-1.11l3.83 3.81c.19.2.45.3.7.3.25 0 .52-.09.7-.3a.996.996 0 0 0 0-1.41v.01zM7 10.7c-2.59 0-4.7-2.11-4.7-4.7 0-2.59 2.11-4.7 4.7-4.7 2.59 0 4.7 2.11 4.7 4.7 0 2.59-2.11 4.7-4.7 4.7z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </form>
  </div>
</DocumentFragment>
`;
